src/app.js

```
@@ -1 +1,70 @@
+/* eslint-disable no-console */
 'use strict';
+
+const fs = require('fs/promises');
+
+async function copy(oldPath, newPath) {
+  if (!oldPath || !newPath) {
+    console.error('Source and destination file paths are required');
+  }
+
+  try {
+    let srcRealPath;
+    let destRealPath;
+
+    try {
+      srcRealPath = await fs.realpath(oldPath);
+    } catch (err) {
+      console.error('Copy error:', err.message);
+      throw err;
+    }
+
+    try {
+      destRealPath = await fs.realpath(newPath);
+    } catch {
+      destRealPath = null;
+    }
+
+    if (destRealPath && srcRealPath === destRealPath) {
+      return;
+    }
+
+    const srcStats = await fs.stat(oldPath);
+
+    if (!srcStats.isFile()) {
+      console.error('Source is not a regular file');
+    }
+
+    if (destRealPath) {
+      const destStats = await fs.stat(newPath);
+
+      if (destStats.isDirectory()) {
+        console.error('Destination is a directory');
+        throw new Error('Destination is a directory');
+      }
+    }
+
+    await fs.copyFile(oldPath, newPath);
+  } catch (err) {
+    console.error('Copy error:', err.message);
+  }
```

The outer catch block inside `copy` logs the error message but does not rethrow it. This silently swallows the error for callers using the exported `copy`. After `console.error` you should rethrow (or throw a new error) so the promise returned by `copy` is rejected and tests/tooling can observe the failure. (Checklist 3.6, 1.4)